Source/MQTTnet/Server/MqttClientSessionsManager.cs

```diff
@@ -224,13 +224,25 @@ async Task TryProcessNextQueuedApplicationMessageAsync(CancellationToken cancell
                     await _retainedMessagesManager.HandleMessageAsync(sender?.ClientId, applicationMessage).ConfigureAwait(false);
                 }
 
+                applicationMessage.DeliveryCount = 0;
+
                 foreach (var clientSession in _sessions.Values)
                 {
                     clientSession.EnqueueApplicationMessage(
                         applicationMessage,
                         sender?.ClientId,
                         false);
                 }
+
+                if (applicationMessage.DeliveryCount == 0)
+                {
+                    if (_options.UndeliveredMessageInterceptor == null)
+                    {
+                        throw new OperationCanceledException(nameof(_options.UndeliveredMessageInterceptor));
```

Comment:
return here dont throw if the user does not care the server should not blow up

Reply:
Makes sense. I wasn't sure how to properly handle this :)

Reply:
This is fixed now.